app/src/main/java/com/example/habitsmasher/ui/dashboard/HabitListFragment.java

```diff
@@ -21,22 +23,36 @@
 import com.firebase.ui.firestore.FirestoreRecyclerOptions;
 import com.google.android.gms.tasks.OnFailureListener;
 import com.google.android.gms.tasks.OnSuccessListener;
+import com.google.android.gms.tasks.Task;
 import com.google.android.material.floatingactionbutton.FloatingActionButton;
+import com.google.firebase.Timestamp;
 import com.google.firebase.firestore.CollectionReference;
+import com.google.firebase.firestore.DocumentReference;
+import com.google.firebase.firestore.DocumentSnapshot;
 import com.google.firebase.firestore.FirebaseFirestore;
+import com.google.firebase.firestore.QuerySnapshot;
 
+import java.sql.Time;
+import java.util.ArrayList;
 import java.util.Date;
 import java.util.HashMap;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
 
-/**
- * The habit list fragment is a container for the list of habits
- */
 public class HabitListFragment extends Fragment {
-    private static final String TAG = "HabitListFragment";
 
+    private static final String TAG = "HabitListFragment";
     private static final HabitList _habitList = new HabitList();
     private HabitItemAdapter _habitItemAdapter;
+    private Button _editButton;
+    private Button _deleteButton;
+    private HabitListFragment _fragment = this;
+
+    // list to store all habitIds for all habits in database
+    private HashSet<Long> _habitIdSet = new HashSet<>();
```

Comment:
If possible, this should be tied into the `HabitList` class, not just a random field in the fragment.